src/link/MachO/load_commands.zig

```diff
@@ -467,8 +467,34 @@ pub inline fn appleVersionToSemanticVersion(version: u32) std.SemanticVersion {
     };
 }
 
-pub fn inferSdkVersionFromSdkPath(path: []const u8) ?std.SemanticVersion {
-    const stem = std.fs.path.stem(path);
+fn readSdkVersionString(arena: Allocator, dir: []const u8) ![]const u8 {
+    const sdk_path = try std.fs.path.join(arena, &.{ dir, "SDKSettings.json" });
+    const contents = try std.fs.cwd().readFileAlloc(arena, sdk_path, std.math.maxInt(u16));
+    const parsed = try std.json.parseFromSlice(std.json.Value, arena, contents, .{});
+    if (parsed.value.object.get("MinimalDisplayName")) |ver| return ver.string;
+    return error.SdkVersionFailure;
+}
+
+pub fn inferSdkVersion(comp: *const Compilation) ?std.SemanticVersion {
+    var arena_allocator = std.heap.ArenaAllocator.init(comp.gpa);
+    defer arena_allocator.deinit();
+    const arena = arena_allocator.allocator();
+    const options = comp.bin_file.options;
+
+    const sdk_dir = switch (options.libc_provider) {
+        .none => unreachable,
+        .installation => unreachable,
```

Comment:
Should this not return `null`? I see in the linker callsites below we don't check for those variants anywhere.

Reply:
I will make a comment about (re)focusing these changes which now reduce to only 1 unreachable case  `.none` which shall only be used when not using a darwin sdk of anykind, and should never be `.none` at this code site. For example, compilation will have it set to `.none` targeting linux.